sdk/python/kfp/compiler/compiler_test.py

```diff
@@ -2889,5 +2889,170 @@ def math_pipeline():
                 add(nums=dsl.Collected(t.output))
 
 
+class TestValidIgnoreUpstreamTaskSyntax(unittest.TestCase):
+
+    def test_basic_permitted(self):
+
+        @dsl.component
+        def fail_op(message: str) -> str:
+            import sys
+            print(message)
+            sys.exit(1)
+            return message
+
+        @dsl.component
+        def print_op(message: str = 'default'):
+            print(message)
+
+        @dsl.pipeline()
+        def my_pipeline(sample_input1: str = 'message'):
+            task = fail_op(message=sample_input1)
+            clean_up_task = print_op(
+                message=task.output).ignore_upstream_failure()
+
+        self.assertEqual(
+            my_pipeline.pipeline_spec.root.dag.tasks['print-op'].trigger_policy
+            .strategy, 2)
+        self.assertEqual(
+            my_pipeline.pipeline_spec.root.dag.tasks['fail-op'].trigger_policy
+            .strategy, 0)
+
+    def test_component_with_no_input_permitted(self):
+
+        @dsl.component
+        def fail_op(message: str) -> str:
+            import sys
+            print(message)
+            sys.exit(1)
+            return message
+
+        @dsl.component
+        def print_op():
+            print('default')
+
+        @dsl.pipeline()
+        def my_pipeline(sample_input1: str = 'message'):
+            task = fail_op(message=sample_input1)
+            clean_up_task = print_op().ignore_upstream_failure()
+
+        self.assertEqual(
+            my_pipeline.pipeline_spec.root.dag.tasks['print-op'].trigger_policy
+            .strategy, 2)
+        self.assertEqual(
+            my_pipeline.pipeline_spec.root.dag.tasks['fail-op'].trigger_policy
+            .strategy, 0)
+
+    def test_clean_up_on_wrapped_pipeline_permitted(self):
+
+        @dsl.component
+        def fail_op(message: str = 'message') -> str:
+            import sys
+            print(message)
+            sys.exit(1)
+            return message
+
+        @dsl.component
+        def print_op(message: str = 'message'):
+            print(message)
+
+        @dsl.pipeline
+        def wrapped_pipeline(message: str = 'message') -> str:
+            task = fail_op(message=message)
+            return task.output
+
+        @dsl.pipeline
+        def my_pipeline(sample_input1: str = 'message'):
+            task = wrapped_pipeline(message=sample_input1)
+            clean_up_task = print_op(
+                message=task.output).ignore_upstream_failure()
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            package_path = os.path.join(tempdir, 'pipeline.yaml')
+            compiler.Compiler().compile(
+                pipeline_func=my_pipeline, package_path=package_path)
+
+        self.assertEqual(
+            my_pipeline.pipeline_spec.root.dag.tasks['print-op'].trigger_policy
+            .strategy, 2)
+        self.assertEqual(
+            my_pipeline.pipeline_spec.root.dag.tasks['fail-op'].trigger_policy
```

Comment:
I believe this should check `wrapped-pipeline`. `fail-op` isn't present in the root dag (`my_pipeline.pipeline_spec.root.dag.tasks`), so the proto is returning the "zero"/default value of `pipeline_spec_pb2.PipelineTaskSpec` and giving a false negative test result because the trigger policy asserted here is the zero/default trigger policy.